### PR TITLE
Update Pop Receipt

### PR DESCRIPTION
**Pull request by wsugarman**
Update the pop receipt for messages once they have been updated.

### CODE REVIEW

**Comment by nytian** [comment]
Pass CI results: :https://durabletaskframework.visualstudio.com/Durable%20Task%20Framework%20CI/_build/results?buildId=6019&view=results

**Comment by cgillum** [comment]
Hi @wsugarman, just to confirm, is this PR ready to be merged?

File: src/DurableTask.AzureStorage/Storage/Queue.cs [hunk 1]
```diff
@@ -59,10 +59,10 @@ await this.queueClient
             this.stats.MessagesSent.Increment();
         }
 
-        public async Task UpdateMessageAsync(QueueMessage queueMessage, TimeSpan visibilityTimeout, Guid? clientRequestId = null, CancellationToken cancellationToken = default)
+        public async Task<UpdateReceipt> UpdateMessageAsync(QueueMessage queueMessage, TimeSpan visibilityTimeout, Guid? clientRequestId = null, CancellationToken cancellationToken = default)
```
Comment (davidmrdavid) [inline src/DurableTask.AzureStorage/Storage/Queue.cs:62]: @wsugarman: is there a reason not to put the the popReceipt-updating logic here (in `UpdateMessageAsync`)? I think I'd rather have it here than sprinkled all throughout the callers, where I think it's more likely for folks to "forget" that they need to capture the output and update the message.

**Review by davidmrdavid — COMMENTED** [review]
Just a small question

**Pull request merged**